src/IB/IBFEMethod.cpp

```diff
@@ -330,6 +330,27 @@ libmesh_restart_file_name(const std::string& restart_dump_dirname,
                      << std::setfill('0') << std::right << time_step_number << "." << extension;
     return file_name_prefix.str();
 }
+
+std::vector<SystemData>
+build_ib_ghosted_system_data(const std::vector<SystemData>& unghosted_system_data, FEDataManager* const fe_data_manager)
+{
+    std::vector<SystemData> ghosted_system_data;
+    EquationSystems* equation_systems = fe_data_manager->getEquationSystems();
+    for (const SystemData& system_data : unghosted_system_data)
+    {
+        const std::string& system_name = system_data.system_name;
+        const System& system = equation_systems->get_system(system_name);
+        NumericVector<double>* original_system_vec = system_data.system_vec;
+        if (!original_system_vec) original_system_vec = system.current_local_solution.get();
+        NumericVector<double>* ghost_system_vec =
+            fe_data_manager->buildGhostedSolutionVector(system_name, /*synch_data*/ false);
```

Comment:
The logic in this function partially duplicates the logic in `LibMeshSystemVectors. Could we set up a private utility function in IBFEMethod that does the same thing here but uses that class to access vectors? That would also allow us to not use `buildGhostedSolutionVector` here, which is a deprecated function.

Reply:
The way that I could see to use `LibMeshSystemVectors` seemed too complicated, and so I just changed this to use `buildIBGhostedVector` and to return a vector of `unique_ptr`s to the built vectors.

Reply:
(One issue is that functions associated with different parts could have different systems associated with them, and so it is not clear that `LibMeshSystemVectors` is the best way to organize things. The `LibMeshSystemVectors` interface is seems best suited for cases where all of the systems do more-or-less the same thing.)

Reply:
Rats. Well, it was worth trying.